providers/yaml/src/main/java/org/jboss/resteasy/plugins/providers/YamlProvider.java

```diff
@@ -52,9 +73,19 @@ public Object readFrom(Class<Object> type, Type genericType, Annotation[] annota
       try {
          LogMessages.LOGGER.debugf("Provider : %s,  Method : readFrom", getClass().getName());
          if (isValidInternalType(type)) {
-            return new Yaml().load(entityStream);
+            // Use the old behavior of trusting everything
+            if (isTypeCheckDisabled()) {
+               return new Yaml().loadAs(entityStream, type);
+            }
+            final BaseConstructor constructor;
+            if (genericType instanceof ParameterizedType) {
+               constructor = new TypeSafeConstructor((ParameterizedType) genericType, getClassLoader(type), allowedList);
+            } else {
+               constructor = new TypeSafeConstructor(type, getClassLoader(type), allowedList);
+            }
+            return new Yaml(constructor).loadAs(entityStream, type);
          } else {
-            CustomClassLoaderConstructor customClassLoaderConstructor = new CustomClassLoaderConstructor(type.getClassLoader());
+            CustomClassLoaderConstructor customClassLoaderConstructor = new CustomClassLoaderConstructor(type, type.getClassLoader());

```

Comment:
Why not a call to getClassLoader() here?

Reply:
I only left it because that's how it 'was", but yeah it likely makes sense to change that too.